src/test/java/uk/gov/hmcts/reform/authorisation/validators/ServiceAuthTokenValidatorTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.reform.authorisation.validators;
+
+import feign.FeignException;
+import feign.Response;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.authorisation.ServiceAuthorisationApi;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+
+import java.util.Collections;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator.BEARER;
+
+public class ServiceAuthTokenValidatorTest {
+    private final ServiceAuthorisationApi api = mock(ServiceAuthorisationApi.class);
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void shouldValidateServiceAuthToken() {
+        final String serviceAuthToken = "service-auth-token";
+        final ServiceAuthTokenValidator validator = new ServiceAuthTokenValidator(api);
+
+        validator.validate(serviceAuthToken);
+        validator.validate(BEARER + serviceAuthToken);
+
+        verify(api, times(2)).authorise(BEARER + serviceAuthToken, new String[0]);
+    }
+
+    @Test
+    public void shouldThrowInvalidTokenException() {
+        exception.expect(InvalidTokenException.class);
+
+        throwFeignException(HttpStatus.UNAUTHORIZED);
+    }
+
+    @Test
+    public void shouldThrowFeignException() {
+        exception.expect(FeignException.class);
```

Comment:
not sure if we should expose this 'internal' exception to the users of this library